js/packages/teams-ai/src/authentication/OAuthBotPrompt.ts

```diff
@@ -0,0 +1,143 @@
+import { OAuthSettings } from './Authentication';
+
+import {
+    ActionTypes,
+    Activity,
+    CardFactory,
+    Channels,
+    InputHints,
+    MessageFactory,
+    OAuthCard,
+    OAuthLoginTimeoutKey,
+    TurnContext
+} from 'botbuilder';
+import * as UserTokenAccess from './UserTokenAccess';
+
+/**
+ * Override the `sendOAuthCard` method to add support for disabling SSO and showing the sign-in card instead.
+ */
+export class OAuthBotPrompt extends OAuthPrompt {
+    private oauthSettings: OAuthSettings;
+
+    constructor(dialogId: string, settings: OAuthSettings) {
+        super(dialogId, settings);
+        this.oauthSettings = settings;
+    }
+
+    async beginDialog(dc: DialogContext): Promise<DialogTurnResult> {
+        // Ensure prompts have input hint set
+        const o: Partial<PromptOptions> = {
+            prompt: {
+                inputHint: InputHints.AcceptingInput
+            },
+            retryPrompt: {
+                inputHint: InputHints.AcceptingInput
+            }
+        };
+
+        // Initialize prompt state
+        const timeout = typeof this.oauthSettings.timeout === 'number' ? this.oauthSettings.timeout : 900000;
+        const state = dc.activeDialog!.state as OAuthPromptState;
+        state.state = {};
+        state.options = o;
+        state.expires = new Date().getTime() + timeout;
+
+        // Prompt user to login
+        await OAuthBotPrompt.sendOAuthCard(this.oauthSettings, dc.context, state.options.prompt);
+        return Dialog.EndOfTurn;
+    }
+
+    /**
+     * Sends an OAuth card.
+     * @param {OAuthPromptSettings} settings OAuth settings.
+     * @param {TurnContext} turnContext Turn context.
+     * @param {string | Partial<Activity>} prompt Message activity.
+     */
+    static override async sendOAuthCard(
+        settings: OAuthSettings,
+        turnContext: TurnContext,
+        prompt?: string | Partial<Activity>
+    ): Promise<void> {
+        // Initialize outgoing message
+        const msg: Partial<Activity> =
+            typeof prompt === 'object'
+                ? { ...prompt }
+                : MessageFactory.text(prompt ?? '', undefined, InputHints.AcceptingInput);
+
+        if (!Array.isArray(msg.attachments)) {
+            msg.attachments = [];
+        }
+
+        // Append appropriate card if missing
+        const msgHasOauthCardAttachment = msg.attachments.some(
+            (a) => a.contentType === CardFactory.contentTypes.oauthCard
+        );
+
+        if (!msgHasOauthCardAttachment) {
+            const cardActionType = ActionTypes.Signin;
+            const signInResource = await UserTokenAccess.getSignInResource(turnContext, settings);
+
+            let link = signInResource.signInLink;
+
+            if (
+                settings.showSignInLink === false ||
+                !this.signInLinkRequiredByChannel(turnContext.activity.channelId)
```

Comment:
Remove extra line. Could you run `yarn llnt` before pushing the next changes up?

Reply:
fixed